src/Stratis.FederatedPeg.Features.FederationGateway/Wallet/FederationWalletManager.cs

```diff
@@ -1021,9 +1043,16 @@ public void ImportMemberKey(string password, string mnemonic, string passphrase)
             this.Wallet.EncryptedSeed = encryptedSeed;
             this.SaveWallet();
 
+            this.isFederationActive = extendedKey.PrivateKey.PubKey.ToHex() == this.federationGatewaySettings.PublicKey;
```

Comment:
The federation only becomes active after `EnableFederation` I think, no need to do it in 2 places.
Also, why do we have 2 calls instead of a single one?
Can we maybe merge what these do into 1 method?

Reply:
Done.